### DISCUSSION TITLE
Using BDFR with Docker / Dockerfile example

### DISCUSSION THREAD

**Discussion by ammacdonald3** [Serene-Arc/bulk-downloader-for-reddit#645] [Q&A]
Thanks for creating such a useful tool!  The app works perfectly for me when installed via pip and run on my terminal, but I'd like to move the code to my Synology NAS through Docker.  After running the below Dockerfile, my terminal updates to show the files being downloaded.  However, I can't ultimately find the downloaded files after the program completes.  Do you by chance have an example of a Dockerfile / run command to show how this should work?  I'm not sure if I'm missing anything about how to convert this for use with Docker. Thanks!

Dockerfile:
`FROM python:3.10-rc-slim`
`RUN python -m pip install \
        bdfr \
        IPTCInfo3`
`CMD ["python", "-m", "bdfr", "download", "./pics-folder", "--subreddit", "pics", "-L", "5", "--file-scheme", "'{SUBREDDIT}--{REDDITOR}--{TITLE}'"]`

Command that I run in terminal:
`docker run -v bdfr-volume:/Users/myusername/bdfr-folder -it --rm bdfr`

**Comment by Serene-Arc** [comment]
Are you familiar with docker? From this question, it doesn't seem like you are, but I'm not as confident with it either. I can create an example Dockerfile soon but at a glance, I don't think you're using the volumes correctly. When you're running the volume option, the first argument is the attachment point inside the docker VM, and the second argument is the attachment in the real filesystem.

So in the command you gave, I believe that you're attaching the folder `bdfr-volume` to the outside of the container, not naming it. Since you're not putting the data there, it won't show anything. Maybe try `docker run -v pics-folder:/Users/myusername/bdfr-folder -it --rm bdfr`. 

If that doesn't work, I'll have to experiment when I'm writing the additional documentation.

**Reply by ammacdonald3** [reply]
Thanks so much for the reply.  You're correct that I do not have too much Docker experience; I've run plenty of images/containers, but I've never created my own like this.  Unfortunately, the suggestion you provided doesn't seem to work either (unless I'm totally missing something on my local machine).  However, I was able to get BDFR working on my NAS by upgrading the Python version there.  Look forward to switching over to Docker at some point in the future once it's added to this project, and thank you again for the help!

**Reply by Serene-Arc** [reply]
Thanks, I'll have a play around. I've gotten docker images working before but it's a tool that I use rarely, so I don't know the syntax and working off hand. I'll add a docker-compose script to the to-do list
